### ISSUE TITLE
Generalisation introduces multiple explicit arguments for one generalisable variable

### ISSUE THREAD

**Issue by nad** [agda/agda#3695]
Agda does not handle the following code correctly:
```agda
{-# OPTIONS --cubical #-}

open import Agda.Builtin.Cubical.Path using (_≡_)
open import Agda.Builtin.Sigma using (Σ; fst)

postulate
  Is-proposition : Set → Set
  subst          : ∀ {A : Set} (P : A → Set) {x y} → x ≡ y → P x → P y

Proposition : Set₁
Proposition = Σ _ Is-proposition

data _/_ (A : Set) (R : A → A → Proposition) : Set where
  [_]  : A → A / R
  resp : ∀ {x y} → fst (R x y) → [ x ] ≡ [ y ]

variable
  A : Set
  R : A → A → Proposition

postulate
  F : (P : A / R → Set)
      (p-[] : ∀ x → P [ x ]) →
      (∀ {x y} (r : fst (R x y)) → subst P (resp r) (p-[] x) ≡ p-[] y) →
      Set
```
The inferred type of `F`:
```agda
{A : Set} (R.1 : (_ _ : A) → Set)
(R.2 : (z z₁ : A) → Is-proposition (R.1 z z₁))
(P : A / (λ z z₁ → R.1 z z₁ Σ., R.2 z z₁) → Set)
(p-[] : (x : A) → P [ x ]) →
({x y : A} (r : R.1 x y) → subst P (resp r) (p-[] x) ≡ p-[] y) →
Set
```
Note that Agda has introduced two arguments corresponding to `R`, and that these arguments are explicit.

**Comment by andreasabel** [comment]
Please keep milestone 2.6.0.1 clear of issues that concern the new features of Agda 2.6.0.

**Comment by UlfNorell** [comment]
The problem is that the meta standing for the generalized variable is eta-expanded. We prevent variable metas from being instantiated by freezing them immediately, but the fix of #658 (cc @andreasabel) ignores freezing for eta expansion.

There shouldn't be any need to eta-expand frozen metas (they will still be equal to their eta-expansion), and not doing it doesn't bring back #658.
